Makefile

```diff
@@ -1628,21 +1635,19 @@ ifeq "$(INSTALL_BYTECODE_PROGRAMS)" "true"
 	  fi; \
 	done
 else
-	for i in $(TOOLS_TO_INSTALL); \
+	for i in $(TOOLS_TO_INSTALL_NAT); \
 	do \
 	  if test -f "tools/$$i".opt$(EXE); then \
 	    $(INSTALL_PROG) "tools/$$i.opt$(EXE)" "$(INSTALL_BINDIR)"; \
 	    (cd "$(INSTALL_BINDIR)" && $(LN) "$$i.opt$(EXE)" "$$i$(EXE)"); \
 	  fi; \
 	done
 endif
+	for i in $(TOOLS_TO_INSTALL_BYT); \
+	do \
+	  $(INSTALL_PROG) "tools/$$i$(EXE)" "$(INSTALL_BINDIR)";\
+	done
```

Comment:
when INSTALL_BYTECODE_PROGRAMS is true, the layout we get is that `foo.byte` and `foo.opt` are available, and `foo` is a symbolic link to `foo.opt`. For bytecode-only program I would expect to have `foo.byte` available, and `foo` as a symbolic link to `foo.byte`. I think that this is not what this line does, it looks like it just installs `foo` and not `foo.byte`. Should we change this?

Reply:
No.  The `.byte` / `.opt` / symbolic link dance is intended to let users choose between the two implementations if they must, and pick the best implementation by default otherwise.  If there's only one implementation, there's no choice to be made.  Plus, there's @kit-ty-kate 's suggestion above to get rid of all this symbolic link nonsense, and I'm in favor.

Reply:
Before this PR, if users for some reason wanted to use a bytecode executable instead of its native counterpart, they would enable INSTALL_BYTECODE_PROGRAMS and then explicitly invoke `foo.byte`. Unless I am missing something, your change breaks this use-case by making `foo.byte` unavailable for some tools. (But then: maybe no one was doing that, and especially for those tools nobody cares about in practice.)

Independently of this backward-compatibility aspect, I don't see the problem with having a symlink. (My Linux box is full of symlinks all over, typically many of the `/usr/lib/*.so` files are symlinks.) It arguably even adds discoverability, if I do `ls -l .../bin/ocamlc` I can immediately tell that I have the native version on my system.

I also didn't interpret Kate's comment https://github.com/ocaml/ocaml/pull/11993#issuecomment-1423088547 to be complaining about symlinks, I read it as suggesting that we install at most one version of each tool, with the absence of symlinks a side-effect of that (or in fact we could keep symlinks if we really wanted, but I agree that in this case it makes less sense).

To summarize: my opinion is still that having the symlinks would be nicer, and I think that they help with compatibility in INSTALL_BYTECODE_PROGRAMS mode. It would probably be okay (not my preference but okay) to do without the links *outside* the INSTALL_BYTECODE_PROGRAMS mode. Then we could discuss (in a separate PR?) disabling this mode by default, and we would get closer to your ideal world (with a configure option still working for `.byte` aficionados).

Reply:
Okay, if you insist, that works for me.